addons/account/models/account_move_line.py

```diff
@@ -576,17 +576,21 @@ def _compute_account_id(self):
                 )[0]
         for line in self:
             if not line.account_id and line.display_type not in ('line_section', 'line_note'):
-                line.account_id = line.company_id.account_journal_suspense_account_id
+                previous_two_accounts = line.move_id.line_ids.filtered(
+                    lambda l: l.account_id and l.display_type == line.display_type
+                )[-2:].account_id
+                if len(previous_two_accounts) == 1:
+                    line.account_id = previous_two_accounts
+                else:
+                    line.account_id = line.company_id.account_journal_suspense_account_id
 
     @api.depends('move_id')
     def _compute_balance(self):
         for line in self:
-            if line.display_type not in ('line_section', 'line_note'):
-                line.balance = (
-                    line.balance
-                    or line.debit - line.credit
-                    or -sum((line.move_id.line_ids - line).mapped('balance'))
-                )
+            if line.display_type not in ('line_section', 'line_note') and not line.move_id.is_invoice():
+                # Only act as a default value when none of balance/debit/credit is specified
+                # balance is always the written field because of `_sanitize_vals`
+                line.balance = -sum((line.move_id.line_ids - line).mapped('balance'))
             else:
                 line.balance = False
```

Comment:
LAS: `False` if it's an invoice? :thinking: 
WAN: But it works!
LAS: How?
WAN: Dark magic!
LAS: :skull:

Reply:
The dark magic is in the ORM: `False` is translated to `0` by it.

I made it more explicit in the code though.